### ISSUE TITLE
A feature list i made for things that could be added

### ISSUE THREAD

**Issue by The-Grace-God** [Beamographic/rush#345]
## As a brief reason why these are not in separate "issues" is because most of these were too small to have as their own or hard to explain without the rest

#Muse Dash features
- Add the ability to swap the backdrop (either set by skin or chart haven't thought much on that yet)
-     Allow charts to be able to choose what backdrops to have at what moment just like how Muse Dash has in this chart here
-     https://youtu.be/hZJaT2gGnhA?si=B4MwMzSYzARRaSzT&t=98
- Shader or FG image support
-     This can be found in a few charts and adds a small effect to gameplay I believe that this could have a really cool use in chart if executed right
-     https://youtu.be/hZJaT2gGnhA?si=B9__HeLMftKS2fYt&t=27
- Add character falling after fail (Lowest priority possible)
-     Really small cosmetic item but could make the gamemode feel more complete
-     https://youtu.be/Vcg7u1MpLms?si=M1npPCfC2F0iKAR7&t=33
- More to be added later as i find more
# Other gimmicks or ideas
- Allow the boss to be changed
-     To allow the boss activated by the chart to be changed via a theme or a loaded in chart theme
- Rolls (ITG)
-     This is a gimmick that i feel could mix up the gameplay a bit im well aware that a system like this exists but that's for minibosses and they take up both lanes my idea would only take one lane similar to a hold and add combo per each press of a button during the duration of the roll (maybe add a bonus for high presses)
- ![roll concept](https://github.com/Beamographic/rush/assets/66783188/29c8c4a2-b0a5-435b-817e-df1720337505)
- Chart defined theme
-     This one pretty much explains itself its just a idea to allow custom charts to use their own theme (possibly add options for what elements the chart can overwrite like backgrounds, characters or the notes)
- Beat based sprites
-     This is a small idea for if the theme chooses to do so instead of animating some elements based by time insteat its set to the current bpm of the song
- More to be added later

**Comment by LumpBloom7** [comment]
Some ideas are covered by #1, but I'll keep this open because it has neat ideas.

**Comment by The-Grace-God** [comment]
@LumpBloom7 
> Some ideas are covered by #1, but I'll keep this open because it has neat ideas.

yeah im not that surprised tbh i was to the other features part of this list when i realized that i may be repeating a idea thats already been done
